R/utils_memoised.R

```diff
@@ -1,28 +1,24 @@
-
-memoise_cran_db <- memoise::memoise({
-  function() {
-    cran <- tools::CRAN_package_db()
-    # remove first instance of column name MD5Sum
-    cran
-  }
-}
-)
-
-
-
 #' Fetch CRAN Mirrors Info
 #'
 #' @param all default \code{TRUE}, passed to \code{\link{utils}[getCRANmirrors]}
-#' @param host_exists whether cran can be contacted
 #' @param ... additional arguments passed to \code{\link{utils}[getCRANmirrors]}
+#' @param .local an optional local directory to source the CRAN package index
+#'   from, defaulting to \code{getOption("riskmetric.tests")}, used
+#'   for isolating repository requests during testing.
 #'
 #' @importFrom curl nslookup
+#' @importFrom memoise memoise
+#'
 memoise_cran_mirrors <- memoise::memoise({
   # add parameter such that memoised results rerun if internet availability changes
   # NOTE: might need to implement actual caching to avoid inconsistent behavior
   # when run with spotty internet
-  function(all = TRUE, ...,
-           host_exists = !is.null(curl::nslookup("cran.r-project.org", error = FALSE))) {
+  function(all = TRUE, ..., .local = getOption("riskmetric.tests")) {
+    if (!is.null(.local)) {
+      return(read.csv(
+        file.path(.local, "test_webmocks", "data", "cran_mirrors.csv"),
```

Comment:
`getOption("riskmetric.tests")` return `NULL` by default. 

We might need to consider if `riskmetric.tests` is clear enough to indicate a path. 

Does  `file.path(.local, "test_webmocks", "data", "cran_mirrors.csv")` always return correct path from user?

Reply:
This option isn't intended to be set by users, but is set only for testing purposes and will always be set to the working directory when tests are run. If you can think of a better way to approach this, I'm open to alternatives and agree that an option isn't ideal. 

https://github.com/pharmaR/riskmetric/blob/84d8810f9ad661db39883578b9cec82c1b3ad883/tests/testthat/setup_mock_web_requests.R#L4

Reply:
Thanks for the clarification.   I am clear of the meaning for `.test` now.